### PR TITLE
remove dependency between workflow bundle and cms

### PR DESCRIPTION
**Pull request by itkg-nanne**

### CODE REVIEW

File: Backoffice/Collector/RoleCollector.php [hunk 1]
```diff
@@ -2,19 +2,49 @@
...
+     */
+    public function __construct(RoleRepositoryInterface $roleRepository, TranslatorInterface $translator, TranslationChoiceManagerInterface $translationChoiceManager)
+    {
+        $this->roleRepository = $roleRepository;
+        $this->translator = $translator;
+        $this->translationChoiceManager = $translationChoiceManager;
+    }
+
+    /**
+     * @param string $role
```
Comment (itkg-alavieille) [inline Backoffice/Collector/RoleCollector.php]: remove this php doc

File: Backoffice/Collector/RoleCollector.php [hunk 2]
```diff
@@ -2,19 +2,49 @@
...
+     */
+    public function loadWorkflowRole()
+    {
+        $workflowRoles = $this->roleRepository->findWorkflowRole();
+        foreach($workflowRoles as $workflowRole) {
```
Comment (itkg-alavieille) [inline Backoffice/Collector/RoleCollector.php]: psr2

File: Backoffice/Collector/RoleCollector.php [hunk 3]
```diff
@@ -2,19 +2,49 @@
...
+        }
+    }
 
     /**
      * @return array
      */
     public function getRoles()
     {
-        return array_unique($this->roles);
+        return $this->roles;
```
Comment (itkg-alavieille) [inline Backoffice/Collector/RoleCollector.php]: why remove array_unique ?

File: Backoffice/Collector/RoleCollector.php [hunk 4]
```diff
@@ -22,6 +52,15 @@ public function getRoles()
...
     public function addRole($role)
     {
-        $this->roles[] = $role;
+        $this->addRoleWithTranslation($role, $this->translator->trans('open_orchestra_role.' . strtolower($role), array(), 'role'));
     }
+
+    /**
+     * @param string $role
+     */
+    public function addRoleWithTranslation($role, $translation)
```
Comment (itkg-alavieille) [inline Backoffice/Collector/RoleCollector.php]: update php doc with translation parameter

File: BackofficeBundle/DependencyInjection/Compiler/AbstractRoleCompilerPass.php [hunk 5]
```diff
@@ -24,5 +24,8 @@ protected function addRoles(ContainerBuilder $container, array $roles = array())
         foreach ($roles as $role) {
             $definition->addMethodCall('addRole', array($role));
         }
+        if($container->hasParameter('open_orchestra_backoffice.collector.role.workflow_role_in_group') && $container->getParameter('open_orchestra_backoffice.collector.role.workflow_role_in_group')) {
```
Comment (itkg-alavieille) [inline BackofficeBundle/DependencyInjection/Compiler/AbstractRoleCompilerPass.php]: ps2

File: BackofficeBundle/Resources/config/collector.yml [hunk 6]
```diff
@@ -1,6 +1,11 @@
 parameters:
     open_orchestra_backoffice.collector.role.class: OpenOrchestra\Backoffice\Collector\RoleCollector
+    open_orchestra_backoffice.collector.role.workflow_role_in_group: true
```
Comment (itkg-alavieille) [inline BackofficeBundle/Resources/config/collector.yml]: add the parameter in configuration with a default not in yml

File: BackofficeBundle/Tests/Collector/RoleCollectorTest.php [hunk 7]
```diff
@@ -38,15 +51,40 @@ public function testAddAndGetRoles(array $newRoles, array $expectedRoles)
...
+     * @param array $newRoles
+     * @param array $expectedRoles
+     *
+     * @dataProvider provideRolesAndExpected
+     */
+    public function testLoadWorkflowRole(array $newRoles, array $expectedRoles)
+    {
+        $roles = new ArrayCollection();
+
+        foreach($newRoles as $newRole) {
```
Comment (itkg-alavieille) [inline BackofficeBundle/Tests/Collector/RoleCollectorTest.php]: ps2

File: BackofficeBundle/Tests/Form/Type/OrchestraRoleChoiceTypeTest.php [hunk 8]
```diff
@@ -78,21 +75,17 @@ public function testBuilder()
      *
      * @dataProvider provideRoleAndTranslation
      */
-    public function testConfigureOptions(array $roles, $translation, array $expectedChoices)
+    public function testConfigureOptions(array $roles)
```
Comment (itkg-alavieille) [inline BackofficeBundle/Tests/Form/Type/OrchestraRoleChoiceTypeTest.php]: update php doc

File: BackofficeBundle/DependencyInjection/Compiler/AbstractRoleCompilerPass.php (hunk 5 above)
Comment (nicolasThal) [inline BackofficeBundle/DependencyInjection/Compiler/AbstractRoleCompilerPass.php]: I would rather let the class decide if she should load the workflow role or not

**Comment by itkg-nanne** [comment]
done @nicolasThal , @itkg-alavieille

File: Backoffice/Collector/RoleCollector.php [hunk 9]
```diff
@@ -2,19 +2,53 @@
...
+        $this->translationChoiceManager = $translationChoiceManager;
+        if ($workflowRoleInGroup) {
+            $this->loadWorkflowRole();
+        }
+    }
+
+    /**
+     * add workflow roles from repository
+     */
+    public function loadWorkflowRole()
```
Comment (nicolasThal) [inline Backoffice/Collector/RoleCollector.php]: this method should be protected

File: Backoffice/Collector/RoleCollector.php [hunk 10]
```diff
@@ -22,6 +56,16 @@ public function getRoles()
...
     {
-        $this->roles[] = $role;
+        $this->addRoleWithTranslation($role, $this->translator->trans('open_orchestra_role.' . strtolower($role), array(), 'role'));
     }
+
+    /**
+     * @param string $role
+     * @param string $translation
+     */
+    public function addRoleWithTranslation($role, $translation)
```
Comment (nicolasThal) [inline Backoffice/Collector/RoleCollector.php]: this method should also be protected

**Comment by itkg-alavieille** [comment]
:+1:

**Pull request merged**